fabric-structure-api-v1/src/main/java/net/fabricmc/fabric/api/structure/v1/FabricStructureBuilder.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.structure.v1;
+
+import java.util.Objects;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import net.minecraft.util.Identifier;
+import net.minecraft.world.gen.GenerationStep;
+import net.minecraft.world.gen.chunk.StructureConfig;
+import net.minecraft.world.gen.chunk.StructuresConfig;
+import net.minecraft.world.gen.feature.ConfiguredStructureFeature;
+import net.minecraft.world.gen.feature.FeatureConfig;
+import net.minecraft.world.gen.feature.StructureFeature;
+
+import net.fabricmc.fabric.impl.structure.FabricStructureUtil;
+import net.fabricmc.fabric.impl.structure.StructuresConfigHooks;
+import net.fabricmc.fabric.mixin.structure.FlatChunkGeneratorConfigAccessor;
+import net.fabricmc.fabric.mixin.structure.StructureFeatureAccessor;
+import net.fabricmc.fabric.mixin.structure.StructuresConfigAccessor;
+
+/**
+ * A builder for registering custom structures.
+ *
+ * <p>Example usage:
+ * <pre>{@code
+ * StructureFeature structure = new MyStructure(DefaultFeatureConfig.CODEC);
+ * ConfiguredStructureFeature<DefaultFeatureConfig, ? extends StructureFeature<DefaultFeatureConfig>> configuredStructure
+ *     = structure.configure(new DefaultFeatureConfig());
+ * FabricStructureBuilder.create()
```

Comment:
Unless I missed something, I think the arguments are missing here:
```suggestion
 * FabricStructureBuilder.create(new Identifier("mymod:mystructure", structure))
```

Reply:
Oops